avocado/plugins/assets.py

```diff
@@ -0,0 +1,245 @@
+        pattern pattern `name = string`.
+        :param node: AST node to be evaluated
+        :type node: ast.*
+        """
+        if isinstance(node.value, ast.Str):
+            # make sure we are into a method, we are not supporting attributes
+            # and module constant assignments at this time
+            if self.current_klass and self.current_method:
+                # variables just to make dictionary assignment line shorter
+                cur_klass = self.current_klass
+                cur_method = self.current_method
+                name = node.targets[0].id
+                self.asgmts[cur_klass][cur_method][name] = node.value.s
+        self.generic_visit(node)
+    # visit_Assign()
+
+    def visit_Call(self, node):  # pylint: disable=C0103
+        """
+        Visit Calls on AST and build list of calls that matches the pattern.
+        :param node: AST node to be evaluated
+        :type node: ast.*
+        """
+        if isinstance(node.func, ast.Attribute):
+            if self.pattern in node.func.attr:
+                call = self._parse_args(node)
+                if call:
+                    self.calls.append(call)
+    # visit_Call()
+
+    def fetch_assets(self):
+        """
+        Fetches the assets based on keywords listed on self.calls.
+        :returns: list of names that were successfuly fetched and list of
+        fails.
+        """
+        cache_dirs = data_dir.get_cache_dirs()
+        success = []
+        fail = []
+        for call in self.calls:
+            expire = call.pop('expire', None)
+            if expire is not None:
+                expire = data_structures.time_to_seconds(str(expire))
+            try:
+                asset.Asset(**call, cache_dirs=cache_dirs,
+                            expire=expire).fetch()
+                success.append(call['name'])
+            except EnvironmentError as failed:
+                fail.append(failed)
+        return success, fail
+    # fetch_assets()
+# FetchAssetHandler()
+
+
+class Assets(CLICmd):
+    """
+    Implements the avocado 'assets' subcommand
+    """
+    name = 'assets'
+    description = 'Manage assets'
+
+    def configure(self, parser):
+        """
+        Add the subparser for the assets action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Assets, self).configure(parser)
+
+        subcommands = parser.add_subparsers(dest='assets_subcommand')
+        subcommands.required = True
+
+        fetch_subcommand_parser = subcommands.add_parser(
+            'fetch',
+            help='Fetch assets from test source or config file if it\'s not'
+            ' already in the cache')
+        fetch_subcommand_parser.add_argument('references', nargs='+',
+                                             metavar='TEST_REFERENCE',
+                                             help='List of test references')
+    # configure()
+
+    def run(self, config):
```

Comment:
It's important for this command to return a meaningful exit code to its callers.  I see this command being used before the `avocado run` invocation, in which a failure to fetch the assets would not attempt to run the tests (for minimum changes of false negatives due to network issues, download timeouts, etc).

Take a look at `avocado.core.exit_codes` for some documented exit statuses.